packages/flutter_tools/lib/src/project_validator.dart

```diff
@@ -11,7 +13,116 @@ abstract class ProjectValidator {
   /// Can return more than one result in case a file/command have a lot of info to share to the user
   Future<List<ProjectValidatorResult>> start(FlutterProject project);
   /// new ProjectValidators should be added here for the ValidateProjectCommand to run
-  static const List <ProjectValidator> allProjectValidators = <ProjectValidator>[
-    // TODO(jasguerrero): add validators
+  static List <ProjectValidator> allProjectValidators = <ProjectValidator>[
+    GeneralInfoProjectValidator(),
   ];
 }
+
+class GeneralInfoProjectValidator extends ProjectValidator{
+  @override
+  Future<List<ProjectValidatorResult>> start(FlutterProject project) async {
+    final YamlMap pubContent = loadYaml(project.pubspecFile.readAsStringSync()) as YamlMap;
```

Comment:
Do we not have elsewhere in the codebase where we parse a project's pubspec? If not, I would recommend pulling this out into its own file. Parsing Yaml is tricky and error prone, we should ideally have a single source code file that knows about the structure of pubspec.yaml and does all the type checking once.

Reply:
In fact, it would be nice if this was a class, that exposed getters for all the fields that we need to read.

Reply:
There is a general package for parsing [pubspec](https://pub.dev/packages/pubspec_parse) but is missing stuff that I need like the flutter key. Agree on making this it's own class

Reply:
Yeah, also we should generally try not to add any more pub dependencies to this repo. We already have a lot of trouble pub solving.